Added area and location

dirz88 opened the pull request:
- Added menu bar for improved user experience (pushy.js)
- Added area: Each macro does have now a location

This is my first project with Node.js or generally in programming. I
hope it meets the requirements and tell me what I need to adjust, it is
adopted.

alexbain commented:
Thanks for the work on this, @dirz88.

Couple thoughts:

1. What if you wanted the same `macro` to work in two different `areas`? Or, what if certain `remotes` were only applicable in certain `areas`? I think it might be more flexible if we introduced `areas` as a top level configuration option which contained an array of `macros` and `remotes` that you wanted to be shown for that `area`.

    Example `config.json`:

    ```
"areas": {
  "Office": {
    "macros": [
      "Listen to Music"
    ],
    "remotes": [
      "Yamaha",
    ]
  },
  "Living Room": {
      "macros": [
        "Listen to Music",
        "Play Xbox 360"
      ],
      "remotes": [
        "TV",
        "Xbox 360",
        "Yamaha"
      ]
  }
}
    ```

2. I believe the two tests are failing because you changed the `macros` route, here: https://github.com/alexbain/lirc_web/pull/22/files#diff-0364f57fbff2fabbe941ed20c328ef1aR124.

3. I like the introduction of a side menu. I'd been thinking about that myself, and pushy.js looks good.

4. If `areas` was a top level configuration option the UI would need to be changed to only show `areas` if that configuration option was set. Otherwise, it would need to only show `macros` and `remotes`

If you're willing to make these changes I'd love to get this added to the project.

dirz88 commented:
Hello @alexbain 

I'll try to integrated your thoughts. At the moment I'm quite busy, but I try to do that as soon as possible :).

dirz88 commented:
I'm sorry, i had some troubles with git to sync/merge. I accidentally requested further commits and then delete the repository :(. Apologies the troubles..

Actually I've changed all to your wishes expect point 4. I don't know how to implement this - still searching for a solution. But I wonder if this is necessary - since at least 1 room is always required.

![main](https://cloud.githubusercontent.com/assets/16446847/12378689/83555780-bd46-11e5-9e9a-28c5736fb194.PNG)
![sidemenu](https://cloud.githubusercontent.com/assets/16446847/12378690/835b42da-bd46-11e5-9915-1b2110c8cfaa.PNG)

I've changed the menubar to show the rooms and their remotes.
I would ask for a new commit, if you agree - or should I try to integrate your point 4 first?

Greetings from switzerland,
dirz88

alexbain commented:
@dirz88 Thanks for updating this. I'll take a look soon and let you any additional thoughts I've got.
